### PR TITLE
Add documentation

### LINKED ISSUE #1: Needs documentation

### PR DESCRIPTION
**Pull request by blacksmithgu**
Fixes #1. This is a nice and simple crate.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1,23 +1,117 @@
...
+    ///
+    /// This method never panics.
     pub fn new(value: T) -> Volatile<T> {
         Volatile(value)
     }
 
+    /// Performs a volatile read of the contained value, returning a copy
+    /// of the read value. Volatile reads are guaranteed not to be optimized
+    /// away by the compiler, but by themselves do not have atomic ordering
+    /// guaruntees. To also get atomicity, consider looking at the `Atomic` wrapper type.
```
Comment (phil-opp) [inline src/lib.rs]: typo: “guaruntees”

**Comment by phil-opp** [comment]
This is really great, thanks a lot!

(I found one small typo, but otherwise this looks good to me.)

**Comment by blacksmithgu** [comment]
Whoops, sorry about that!

**Comment by phil-opp** [comment]
Thanks so much!

**Pull request merged**